compiler/semstmts.nim

```diff
@@ -729,6 +729,16 @@ proc semRaise(c: PContext, n: PNode): PNode =
     var typ = n.sons[0].typ
     if typ.kind != tyRef or typ.lastSon.kind != tyObject:
       localError(n.info, errExprCannotBeRaised)
+    
+    # check if the given object inherits from Exception
+    var base = typ.lastSon
+    while true:
+      if base.sym.name.s == "Exception":
```

Comment:
That is not enough, you need to at least also check ``base.owner.flags.contains(sfSystemModule)``. But the clean solution is to mark ``Exception`` with a new magic that you check for here.

Reply:
Okay, is there any good documentation of how to use magic in the compiler or could you point me to a file, where this is used extensively?

Reply:
add 'mException' to the TMagic enum in ast.nim and check for it here. Mark Exception as ``magic: Exception``

Reply:
Tried this, but I can get the magic mException to work on Exception. I added it to TMagic and can use it in code (tried echo), but when I write `{.magic: "Exception", compilerproc.}` or `{.magic Exception, compilerproc.}` i get *Error: type expected* immediatly on this line.

Reply:
Tried this, but I can't get the magic mException to work on Exception. I added it to TMagic and can use it in code (tried echo), but when I write `{.magic: "Exception", compilerproc.}` or `{.magic Exception, compilerproc.}` on Exception, I get *Error: type expected* immediatly on this line.

Reply:
For me 
```nim

  Exception* {.compilerproc, magic: "Exception".} = object of RootObj
```
works.

Reply:
But it doesn't work for me :/

I've added `mException` to TMagic in ast.nim.
Than I ran `.\koch boot` which compiled fine and in this state I can use `TMagic.mException`.
Than I copy pasted your line and run `.\koch boot` and I get the aformentioned error.

The error is 

    > lib\system.nim(466, 3) Error: type expected

so the error doesn't complain about the pragma directly, but rather about `Exception` as a type, but weirdly when I remove the `magic` pragma the message disapears